apps/user/src/components/form/지원자정보/ProfileUpload/ProfileUpload.tsx

```diff
@@ -0,0 +1,97 @@
+import { color, font } from '@maru/theme';
+import { Button, Column } from '@maru/ui';
+import {
+    useImageFileChange,
+    useImageFileDragAndDrop,
+    useOpenUploadImageFile,
+    useUploadProfileImageFile,
+} from './ProfileUpload.hooks';
+import styled from 'styled-components';
+import { useState } from 'react';
+
+const ProfileUpload = () => {
+    const [profileImage, setProfileImage] = useState('');
+
+    const { uploadProfileImageFile } = useUploadProfileImageFile(setProfileImage);
+    const { imageFileInputRef, handleImageUploadButtonClick } = useOpenUploadImageFile();
+    const { isDragging, onDragEnter, onDragLeave, onDragOver, onDrop } =
+        useImageFileDragAndDrop(uploadProfileImageFile);
+    const { handleImageFileChange } = useImageFileChange(uploadProfileImageFile);
+
+    return (
+        <StyledProfileUpload>
+            <Title>증명사진</Title>
+            {profileImage ? (
+                <ImagePreview src={profileImage} alt="profile-image" />
+            ) : (
+                <ImageUploadBox
+                    onDragEnter={onDragEnter}
+                    onDragLeave={onDragLeave}
+                    onDragOver={onDragOver}
+                    onDrop={onDrop}
+                    isDragging={isDragging}>
```

Comment:
제 생각엔 isDragging, onDragEnter 등등 이런 것들을 구조 분해할 필요가 없다고 생각해요.
왜냐하면 이 훅은 이미지 파일을 드래그 앤 드랍할 수 있게 해준다는게 목적인데
Drag and drop에 쓰이는 이런 세부적인 속성을 드러내 관심을 끌 필요가 없다고 생각하기 때문입니다.

```suggestion
    const imageFileDragAndDrop =
        useImageFileDragAndDrop(uploadProfileImageFile);
    const { handleImageFileChange } = useImageFileChange(uploadProfileImageFile);

    return (
        <StyledProfileUpload>
            <Title>증명사진</Title>
            {profileImage ? (
                <ImagePreview src={profileImage} alt="profile-image" />
            ) : (
                <ImageUploadBox
                    {...imageFileDragAndDrop}
                >
```

Reply:
제 생각은 좀 다른데요.. 저런식으로 ... 으로 모든 props를 넘겨주었을때  어디서 정의되고 어떤 프롭스를 포함하는지에 대한 정보가 명확하지않아 별로라고 생각합니다 좀 더 명확하고 어떤 props들을 넘겨주는지 아는게 그 동작에 대해서도 이해가 쉬울것이라고 생각합니다

Reply:
저는 그 컴포넌트에 세부적이거나 부가적인 동작들은 ... 을 쓰는게 좋다고 생각하지만 핵심 동작들은 보여주는게 좋다고 생각합니다 명재님은 어떻게 생각하시나요?

Reply:
그렇긴 하네요. 다시 생각해보니 스프레드 프롭스 이점이 프롭스 타입을 명시할 필요없는 자바스크립트에서만 작용할 것 같다고 생각이 드네요.